micro-rdk/src/common/data_manager.rs

```diff
@@ -361,7 +415,7 @@ where
         Ok(msg)
     }
 
-    async fn run<'b>(&mut self, app_client: &'b AppClient) -> Result<(), AppClientError> {
+    async fn run<'b>(&self, app_client: &'b AppClient) -> Result<(), AppClientError> {
```

Comment:
asking for curiosity: does `self` not need to be a mutable reference to get the store lock below? Was that using some sort of interior mutability this whole time?

Reply:
you can usually get a mutable reference from an &self after locking. I haven't tried the data upload yet so there might be a potential issue here indeed